### ISSUE TITLE
ModuleNotFoundError: No module named 'click.shell_completion'

### ISSUE THREAD

**Issue by garystafford** [sqlfluff/sqlfluff#2323]
## Observed Behaviour
`pip install sqlfluff==0.9.0`
`sqlfluff --help`
`sqlfluff --version`
`python3 -m sqlfluff lint --dialect redshift --ignore parsing,templating dags/sql_redshift/ --format github-annotation`

All commands are working as expected with no errors.

`pip install sqlfluff==0.9.1`
`sqlfluff --help`
`sqlfluff --version`
`python3 -m sqlfluff lint --dialect redshift --ignore parsing,templating dags/sql_redshift/ --format github-annotation`
`sqlfluff lint --dialect hive --ignore parsing,templating dags/sql_data_lake/`

Same error for all commands, using version 0.9.1:

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/sqlfluff/__main__.py", line 2, in <module>
    from sqlfluff.cli.commands import cli
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/sqlfluff/cli/commands.py", line 28, in <module>
    from sqlfluff.cli.autocomplete import dialect_shell_complete
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/sqlfluff/cli/autocomplete.py", line 3, in <module>
    from click.shell_completion import CompletionItem
ModuleNotFoundError: No module named 'click.shell_completion'
```

## Steps to Reproduce
`pip install sqlfluff==0.9.1`
`sqlfluff --help`
`sqlfluff --version`

## Dialect
It throws an error for any dialect.

## Version
```
> sqlfluff --version

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/bin/sqlfluff", line 5, in <module>
    from sqlfluff.cli.commands import cli
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/sqlfluff/cli/commands.py", line 28, in <module>
    from sqlfluff.cli.autocomplete import dialect_shell_complete
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/sqlfluff/cli/autocomplete.py", line 3, in <module>
    from click.shell_completion import CompletionItem
ModuleNotFoundError: No module named 'click.shell_completion'
```

## Configuration
```
> python --version
Python 2.7.16

> python3 --version
Python 3.8.0
```

**Comment by jpy-git** [comment]
Hi @garystafford, this is likely because the version of `click` you have installed is older than `8.0` so is missing that `click.shell_completion` module that we use. When you updated SQLFluff it didn't update the required click version since we weren't aware of this at the time and didn't place the constraint on the click requirement.

We had a PR the other day (#2271) to ensure that on subsequent releases your click will be upgraded to the right version.

In the meantime running `pip install -U click` should clear this up 👍

**Comment by garystafford** [comment]
Thanks for the quick feedback, @jpy-git. That worked. I updated `click`, which allowed me to use the latest version of `sqlfluff`: `pip install sqlfluff==0.9.1 click==8.0.3`.

**Comment by jpy-git** [comment]
Fab, glad to hear that and credit to @wong-codaio for pointing out the issue in the first place 😄
